src/utils/helpers/buildMapper.ts

```diff
@@ -40,51 +42,34 @@ const dateToTimezone = (date: string) => {
   });
 };
 
-export const buildMapper = (data: Build[], currentBuildId: number) => {
-  return data.map(
-    ({ id, status, status_info, ended_on, scheduled_on }: Build) => {
-      let duration = 0;
-      if (ended_on && scheduled_on) {
-        const startTime = new Date(scheduled_on);
-        const endTime = new Date(ended_on);
-        duration = (endTime.valueOf() - startTime.valueOf()) / 60000;
-        duration = Math.round(duration);
-      }
-      if (id === currentBuildId) {
-        return {
-          id,
-          name: `${dateToTimezone(ended_on ?? scheduled_on)} - Active`,
-          status: isCompleted(status, duration),
-          status_info
-        };
-      }
-
-      if (isBuilding(status)) {
-        return {
-          id,
-          name: `${dateToTimezone(scheduled_on)} - Building`,
-          status: "Building",
-          status_info
-        };
-      }
-
-      if (isQueued(status)) {
-        return {
-          id,
-          name: `${dateToTimezone(scheduled_on)} - Queued`,
-          status: "Building",
-          status_info
-        };
-      }
+export const buildDatetimeStatus = (
+  { id, status, ended_on, scheduled_on }: Build,
+  currentBuildId: number
+): string => {
+  if (id === currentBuildId) {
+    return `${dateToTimezone(ended_on ?? scheduled_on)} - Active`;
+  } else if (status === "BUILDING") {
+    return `${dateToTimezone(scheduled_on)} - Building`;
+  } else if (status === "QUEUED") {
+    return `${dateToTimezone(scheduled_on)} - Queued`;
+  } else {
+    return `${dateToTimezone(ended_on ?? scheduled_on)} - ${
+      STATUS_OPTIONS[status]
+    }`;
+  }
+};
 
-      return {
-        id,
-        name: `${dateToTimezone(ended_on ?? scheduled_on)} - ${
-          STATUS_OPTIONS[status]
-        }`,
-        status: isCompleted(status, duration),
-        status_info
-      };
+export const buildStatus = ({
```

Comment:
Do we need to combine the `BUILDING` and `QUEUED` statuses? It feels like it would be useful to have these distinct states represented distinctly in the UI.

Reply:
Update. I asked @kcpevey on Slack and she said it looks like a mistake, so I've changed it and we'll see if anybody complains or it breaks anything.